sources/packages/api/src/auth/jwt.strategy.ts

```diff
@@ -32,16 +32,18 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
     // In the future we can decide how to proper handle roles, but the only issue to have
     // then flatten for now is if we create a role with the same name in 2 different
     // clients on Keycloak. For now we just have the Student client and in the future
-    // it would be pretty easy to adpat this method as needed to expose the roles
+    // it would be pretty easy to adapt this method as needed to expose the roles
     // also as needed to be validate using a decorator/annotation.
     userToken.roles = [];
-    Object.keys(payload.resource_access).forEach((value) => {
-      if (Array.isArray(payload.resource_access[value].roles)) {
-        payload.resource_access[value].roles.forEach((roleValue: string) => {
-          userToken.roles.push(roleValue);
-        });
-      }
-    });
+    if (payload.resource_access) {
+      Object.keys(payload.resource_access).forEach((value) => {
+        if (Array.isArray(payload.resource_access[value].roles)) {
```

Comment:
Why are we checking for array type? is there a possibility that we may get a non-array type here?

Reply:
Apparently no. I removed it and executed the E2E tests and all passed 😉